tests/backup/backup_test.go

```diff
@@ -1129,6 +1129,214 @@ var _ = Describe("{ShareBackupAndEdit}", func() {
+				log.InfoD("Created Backup Location with name - %s", BackupLocationName)
+			}
+		})
+		Step("Register source and destination cluster for backup", func() {
+			log.InfoD("Registering Source and Destination clusters and verifying the status")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			CreateSourceAndDestClusters(orgID, "", "", ctx)
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster status")
+		})
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			var sem = make(chan struct{}, 10)
+			var wg sync.WaitGroup
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range bkpNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					sem <- struct{}{}
+					time.Sleep(3 * time.Second)
+					backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+					backupNames = append(backupNames, backupName)
+					wg.Add(1)
+					go func(backupName string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						defer func() { <-sem }()
+						CreateBackup(backupName, SourceClusterName, BackupLocationName, backupLocationUID, []string{namespace},
+							nil, orgID, clusterUid, "", "", "", "", ctx)
+					}(backupName)
+				}
+				wg.Wait()
+			}
+			log.Infof("List of backups - %v", backupNames)
+		})
+		Step("Share backup with multiple users", func() {
+			log.InfoD("Share backup with multiple users")
+			log.Infof("Sharing backup with  multiple users")
+
+			// Get Admin Context - needed to share backup and get backup UID
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			// Share backups with aal the users
+			for _, backup := range backupNames {
+				err = ShareBackup(backup, nil, users, ViewOnlyAccess, ctx)
+				log.FailOnError(err, "Failed to share backup %s", backup)
+			}
+
+			for _, user := range users {
+				// Get user context
+				ctxNonAdmin, err := backup.GetNonAdminCtx(user, "Password1")
+				log.FailOnError(err, "Fetching px-central-admin ctx")
+				userContexts = append(userContexts, ctxNonAdmin)
+
+				for _, backup := range backupNames {
+					// Get Backup UID
+					backupDriver := Inst().Backup
+					backupUID, err := backupDriver.GetBackupUID(ctx, backup, orgID)
+					log.FailOnError(err, "Failed while trying to get backup UID for - %s", backup)
+
+					// Delete backup to confirm that the user cannot delete the backup
+					_, err = DeleteBackup(backup, backupUID, orgID, ctxNonAdmin)
+					log.Infof("Error message - %s", err.Error())
+					dash.VerifyFatal(strings.Contains(err.Error(), "doesn't have permission to delete backup"), true, "Verifying backup deletion is not possible")
```

Comment:
Also need to verify that the users cannot restore the backup with view only access

Reply:
Done